src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,131 @@
 package core.basesyntax;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int DEFAULT_CAPACITY = 16;
+    private int threshold = (int) (LOAD_FACTOR * DEFAULT_CAPACITY);
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void put(K key, V value) {
-
+        if (size++ > threshold) {
+            resize();
+        }
+        int index = getIndex(key);
+        Node<K, V> node = table[index];
+        if (table[index] == null) {
+            table[index] = new Node<>(key, value, null);
+        } else {
+            while (true) {
+                if (Objects.equals(key, node.key)) {
+                    node.value = value;
+                    size--;
+                    return;
+                }
+                if (node.next == null) {
+                    break;
+                }
+                node = node.next;
+            }
+            node.next = new Node<>(key, value, null);
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        int index = getIndex(key);
+        if (table[index] == null) {
+            return null;
+        } else {
+            Node<K, V> node = table[index];
+            while (true) {
+                if (Objects.equals(key, node.key)) {
+                    return node.value;
+                }
+                if (node.next == null) {
+                    return null;
+                }
+                node = node.next;
+            }
```

Comment:
avoid while(true)
whole getvalue method can be done with 1 for loop and if check